function/templates/ScaledObject.yaml

```diff
@@ -1,8 +1,9 @@
-{{- if eq .Values.trigger.blob.connection "" }}
+{{- $authRequired := include "function.auth-required" . -}}
```

Comment:
Still have a feeling that this is complicated, not sure if having a separate triggerAuthentication list will make it any better.

```suggestion
{{- $authRequired := include "function.auth-required" . -}}
```

Reply:
@timja any thoughts on this ?

Reply:
could make it a configuration value rather than detecting it? if that's the issue you're raising

Reply:
yeah, that's what i feel will be simpler especially because trigger auth has more scope to grow.
https://keda.sh/docs/1.4/concepts/authentication/#re-use-credentials-and-delegate-auth-with-triggerauthentication